tools/jsrun.py

```diff
@@ -33,7 +35,47 @@ def make_command(filename, engine=None, args=[]):
   # Use "'d8' in" because the name can vary, e.g. d8_g, d8, etc.
   return engine + [filename] + (['--expose-wasm', '--'] if 'd8' in jsengine or 'jsc' in jsengine else []) + args
 
-def run_js(filename, engine=None, args=[], check_timeout=False, stdin=None, stdout=PIPE, stderr=None, cwd=None, full_output=False, assert_returncode=0, error_limit=-1):
+
+def check_engine(engine, rootpath):
```

Comment:
instead of passing in rootpath, you can find it by doing `os.path.dirname(os.path.dirname(__file__))` (to get dir of this file, then one up).